pkg/promotion/env_location.go

```diff
@@ -0,0 +1,43 @@
+package promotion
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/rhd-gitops-example/services/pkg/util"
+)
+
+type EnvLocation struct {
+	RepoPath string // URL or local path
+	Branch   string
+	Folder   string
+}
+
+func (env EnvLocation) IsLocal() bool {
+	parsed, err := url.Parse(env.RepoPath)
+	if err != nil {
+		log.Printf("Error while parsing URL for environment path: '%v', assuming it is local.\n", err)
+		return true
+	}
+	return parsed.Scheme == ""
+}
+
+// String implements Stringer so that messages (including logs, commits and PRs)
+// can reference EnvLocations consistently and in a more readable way.
+func (env EnvLocation) String() string {
+	if env.IsLocal() {
+		return "local filesystem directory"
+	}
+	var b strings.Builder
+	_, shortname, err := util.ExtractUserAndRepo(env.RepoPath)
+	if err != nil {
+		shortname = env.RepoPath
+	}
```

Comment:
And if it is nil? I'm wondering if this method should return an error too

Reply:
Great point... The main point of this function is pretty-printing, and I wanted it to work with `%v` in `fmt`, which means I can't add an error to the function (it wouldn't satisfy the Stringer interface).

Also this function will have to change to deal with the case where `IsLocal` returns an error now. In the error cases, I'll have it use the full RepoPath for now, but I'm open to other suggestions.

Reply:
New version up, let me know what you think of the change